pkg/generic/thrift/http.go

```diff
@@ -77,12 +113,90 @@ func (r *ReadHTTPResponse) SetBase64Binary(enable bool) {
 	r.base64Binary = enable
 }
 
+// SetUseRawBodyForHTTPResp ...
+func (r *ReadHTTPResponse) SetUseRawBodyForHTTPResp(useRawBodyForHTTPResp bool) {
+	r.useRawBodyForHTTPResp = useRawBodyForHTTPResp
+}
+
+// SetDynamicGo ...
+func (r *ReadHTTPResponse) SetDynamicGo(convOpts *conv.Options, msg remote.Message) {
+	r.t2jBinaryConv = t2j.NewBinaryConv(*convOpts)
+	r.msg = msg
+	r.dynamicgoEnabled = true
+}
+
 // Read ...
 func (r *ReadHTTPResponse) Read(ctx context.Context, method string, in thrift.TProtocol) (interface{}, error) {
+	// fallback logic
+	if !r.dynamicgoEnabled {
+		return r.originalRead(ctx, method, in)
+	}
+
+	// dynamicgo logic
+	if r.msg.PayloadLen() == 0 {
+		return nil, perrors.NewProtocolErrorWithMsg("msg.PayloadLen should always be greater than zero")
+	}
+
+	tProt, ok := in.(*cthrift.BinaryProtocol)
+	if !ok {
+		return nil, perrors.NewProtocolErrorWithMsg("TProtocol should be BinaryProtocol")
+	}
+	mBeginLen := bthrift.Binary.MessageBeginLength(method, thrift.TMessageType(r.msg.MessageType()), r.msg.RPCInfo().Invocation().SeqID())
+	sName, err := in.ReadStructBegin()
+	if err != nil {
+		return nil, err
+	}
+	sBeginLen := bthrift.Binary.StructBeginLength(sName)
+	fName, typeId, id, err := in.ReadFieldBegin()
+	if err != nil {
+		return nil, err
+	}
+	fBeginLen := bthrift.Binary.FieldBeginLength(fName, typeId, id)
+	transBuf, err := tProt.ByteBuffer().ReadBinary(r.msg.PayloadLen() - mBeginLen - sBeginLen - fBeginLen - bthrift.Binary.MessageEndLength())
+	if err != nil {
+		return nil, err
+	}
+	fid := dthrift.FieldID(id)
+
+	resp := descriptor.NewHTTPResponse()
+	ctx = context.WithValue(ctx, conv.CtxKeyHTTPResponse, resp)
+	fnDsc := r.svc.DynamicGoDsc.Functions()[method]
+	if fnDsc == nil {
+		return nil, fmt.Errorf("missing method: %s in service: %s in dynamicgo", method, r.svc.DynamicGoDsc.Name())
+	}
+	tyDsc := fnDsc.Response()
+	// json size is usually 2 times larger than equivalent thrift data
+	buf := make([]byte, 0, len(transBuf)*2)
+
+	for _, field := range tyDsc.Struct().Fields() {
+		if fid == field.ID() {
+			// decode with dynamicgo
+			// thrift []byte to json []byte
+			if err = r.t2jBinaryConv.DoInto(ctx, field.Type(), transBuf, &buf); err != nil {
```

Comment:
according to @AsterDY , we should use `Do` here since we don't know the exact length needed

Reply:
DoInto will resize buf if actual size is larger, no problem here.